doc/build-osx.md

```diff
@@ -1,116 +1,316 @@
 
-Then install [Homebrew](https://brew.sh).
+### 2. Homebrew Package Manager
 
-## Dependencies
-```shell
-brew install automake libtool boost miniupnpc libnatpmp pkg-config python qt libevent qrencode
+Homebrew is a package manager for macOS that allows one to install packages from the command line easily.
+While several package managers are available for macOS, this guide will focus on Homebrew as it is the most popular.
+Since the examples in this guide walking through the installation of a package will use Homebrew, it is recommended that you install it to follow along.
+
+To install the Homebrew package manager (if not already installed), run the following command from your terminal:
+
+``` bash
+/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
 ```
 
-If you run into issues, check [Homebrew's troubleshooting page](https://docs.brew.sh/Troubleshooting).
-See [dependencies.md](dependencies.md) for a complete overview.
+Note: If you run into issues while installing homebrew or pulling packages, refer to [Homebrew's troubleshooting page](https://docs.brew.sh/Troubleshooting).
 
-If you want to build the disk image with `make deploy` (.dmg / optional), you need RSVG:
-```shell
-brew install librsvg
+### 3. Install Required Dependencies
+
+The first step is to download the required dependencies.
+These dependencies represent the packages required to get a barebones installation up and running.
+To install, run the following from your terminal:
+
+``` bash
+brew install automake libtool boost pkg-config libevent
 ```
 
-and [`macdeployqtplus`](../contrib/macdeploy/README.md) dependencies:
-```shell
-pip3 install ds_store mac_alias
+### 4. Clone Bitcoin repository
+
+Now that `git` and all the required dependencies are installed, let's clone the Bitcoin Core repository to a directory.
+All build scripts and commands will run from this directory.
+
+``` bash
+git clone https://github.com/bitcoin/bitcoin.git
 ```
 
-The wallet support requires one or both of the dependencies ([*SQLite*](#sqlite) and [*Berkeley DB*](#berkeley-db)) in the sections below.
-To build Bitcoin Core without wallet, see [*Disable-wallet mode*](#disable-wallet-mode).
+### 5. Install Optional Dependencies
 
-#### SQLite
+#### Wallet Dependencies
 
-Usually, macOS installation already has a suitable SQLite installation.
-Also, the Homebrew package could be installed:
+It is not necessary to build wallet functionality to run bitcoind or the GUI.
+To enable legacy wallets, you must install `berkeley-db4`.
+To enable [descriptor wallets](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md), `sqlite` is required.
+Skip `berkeley-db4` if you intend to *exclusively* use descriptor wallets.
 
-```shell
+###### Legacy Wallet Support
+
+`berkeley-db4` is required to enable support for legacy wallets.
+Skip if you don't intend to use legacy wallets.
+
+``` bash
+brew install berkeley-db4
```

Comment:
Package name `berkeley-db4` seems [old](https://github.com/Homebrew/formulae.brew.sh/blob/master/_data/formula/berkeley-db%404.json). Maybe use (here and other places) `berkeley-db@4`?

Reply:
You're right: https://formulae.brew.sh/formula/berkeley-db@4

Using the `Formerly Known as` name might not be so safe